### ISSUE TITLE
Building Enunciate: package genealogy.client.cite does not exist

### ISSUE THREAD

**Issue by sg-ad** [stoicflame/enunciate#232]
When trying to build the project (using `mvn install`, for example), I get an error about a missing package:
```
package genealogy.client.cite does not exist
```
In fact, all `genealogy.*` packages are missing. From what I can see, this package is actually created on the fly from `com.webcohesion.enunciate.examples.cxf.genealogy`.  However, I'm not sure how this gets triggered.  Any tips?
Thanks!

**Comment by stoicflame** [comment]
I can't figure out how you're getting that error.

You're just doing a "mvn install" from the root directory?

**Comment by sg-ad** [comment]
That's right.  I get this error trying to build on `master`.  Here's a bit more context:
```
dev/enunciate/examples/full-api-edge-cases/src/test/java/com/webcohesion/enunciate/tests/jaxwsrijersey/TestFullJAXWSRIAPIWithJAXWSClient.java:[20,29] package genealogy.client.cite does not exist
```
If I skip over the `cxf`, `full-api-edge-cases` and `jboss` modules then it builds fine (and probably skips over a ton of useful tests :) ).

I noticed that in the "Building Enunciate", you mention the need for some "unusual" libraries in order to run the tests.  I'm pretty sure I have many of these already installed, but could a missing dependency be the cause?

**Comment by stoicflame** [comment]
Yes, there's a bunch of libraries that need to be installed in order for the "full" build to run. These include things like mono, ruby, php, etc. By default, the build assumes you _don't_ have these things installed, skips all those tests, and won't let you do `mvn deploy`.

On my box, I have my `~/.m2/settings.xml` file providing all the needed hooks for these things, including setting the `activate.full.tests` property to `true`, which enables the tests and allows me to deploy. I run on Linux. Theoretically, the "full" build should be able to run on Windows and Mac, but I've never done it personally.

So the issue just means that I need to fix the build so that it won't try to compile the tests unless the `activate.full.tests` property is `true`. I can fix that.

Here are the relevant parts of my `~/.m2/settings.xml` file:

```xml
<settings xmlns="http://maven.apache.org/POM/4.0.0"
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/settings-1.0.0.xsd">
  <profiles>
    <profile>
      <id>enunciate-activate-full-tests</id>
      <activation>
        <activeByDefault>true</activeByDefault>
      </activation>
      <properties>
        <activate.full.tests>true</activate.full.tests>
      </properties>
    </profile>
    <profile>
      <id>mono-installed</id>
      <activation>
        <activeByDefault>true</activeByDefault>
      </activation>
      <properties>
        <gmcs.executable>gmcs</gmcs.executable>
      </properties>
    </profile>
    <profile>
      <id>ruby-installed</id>
      <activation>
        <activeByDefault>true</activeByDefault>
      </activation>
      <properties>
        <ruby.executable>ruby</ruby.executable>
      </properties>
    </profile>
    <profile>
      <id>php-installed</id>
      <activation>
        <activeByDefault>true</activeByDefault>
      </activation>
      <properties>
        <php.executable>php</php.executable>
      </properties>
    </profile>
    <profile>
      <id>gcc-installed</id>
      <activation>
        <activeByDefault>true</activeByDefault>
      </activation>
      <properties>
        <gcc.executable>gcc</gcc.executable>
        <libxml.config.executable>xml2-config</libxml.config.executable>
        <gnustep.config.executable>gnustep-config</gnustep.config.executable>
      </properties>
    </profile>
  </profiles>
  ...
</settings>
```

**Comment by scoquelin** [comment]
FYI, same issue for me building master on Linux using `mvn clean install`
Here is the log : 
````
[INFO] 72 errors 
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Enunciate - Parent ................................. SUCCESS [  1.116 s]
[INFO] Enunciate - Core Annotations ....................... SUCCESS [  1.299 s]
[INFO] Enunciate - Javac Support .......................... SUCCESS [  1.065 s]
[INFO] Enunciate - Core Build-time Support ................ SUCCESS [  3.448 s]
[INFO] Enunciate - Runtime Utilities ...................... SUCCESS [  1.030 s]
[INFO] Enunciate - JAXB XML Model Support ................. SUCCESS [  1.044 s]
[INFO] Enunciate - Jackson 2.x JSON Model Support ......... SUCCESS [  0.741 s]
[INFO] Enunciate - Jackson 1.x JSON Model Support ......... SUCCESS [  0.652 s]
[INFO] Enunciate - JAX-RS Model Support ................... SUCCESS [  0.565 s]
[INFO] Enunciate - JAX-WS Model Support ................... SUCCESS [  0.589 s]
[INFO] Enunciate - Spring Web Model Support ............... SUCCESS [  0.591 s]
[INFO] Enunciate - IDL Module ............................. SUCCESS [  4.258 s]
[INFO] Enunciate - Java XML Client Module ................. SUCCESS [  4.956 s]
[INFO] Enunciate - Java JSON Client Module ................ SUCCESS [  3.834 s]
[INFO] Enunciate - C# XML Client Module ................... SUCCESS [  0.567 s]
[INFO] Enunciate - C XML Client Module .................... SUCCESS [  0.563 s]
[INFO] Enunciate - Objective C XML Client Module .......... SUCCESS [  0.528 s]
[INFO] Enunciate - PHP XML Client Module .................. SUCCESS [  0.487 s]
[INFO] Enunciate - PHP JSON Client Module ................. SUCCESS [  0.632 s]
[INFO] Enunciate - Ruby JSON Client Module ................ SUCCESS [  0.532 s]
[INFO] Enunciate - GWT JSON Overlay Module ................ SUCCESS [  3.718 s]
[INFO] Enunciate - Swagger Module ......................... SUCCESS [  0.622 s]
[INFO] Enunciate - Docs Module ............................ SUCCESS [  0.500 s]
[INFO] Enunciate - Slim Maven Plugin ...................... SUCCESS [  1.824 s]
[INFO] Enunciate - Top .................................... SUCCESS [  0.767 s]
[INFO] Enunciate - Maven Plugin ........................... SUCCESS [  2.534 s]
[INFO] Enunciate - Simple Archetype ....................... SUCCESS [  0.602 s]
[INFO] Enunciate - Example Projects ....................... SUCCESS [  0.151 s]
[INFO] Enunciate - Full API With Edge Cases Example ....... FAILURE [  0.438 s]
[INFO] Enunciate - Jackson 2 API Example .................. SKIPPED
[INFO] Enunciate - Jersey Storage Example ................. SKIPPED
[INFO] Enunciate - Spring-Managed Jersey Storage Example .. SKIPPED
[INFO] Enunciate - JBoss Full API Example ................. SKIPPED
[INFO] Enunciate - CXF Full API Example ................... SKIPPED
[INFO] Enunciate - Spring-Web Petclinic Example ........... SKIPPED
[INFO] Enunciate - Contract-First API Example ............. SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 40.787 s
[INFO] Finished at: 2015-12-22T11:39:36-05:00
[INFO] Final Memory: 94M/1151M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.3:testCompile (default-testCompile) on project enunciate-examples-full-api-edge-cases: Compilation failure: Compilation failure:
[ERROR] /home/seb/git/enunciate/examples/full-api-edge-cases/src/test/java/com/webcohesion/enunciate/tests/jaxwsrijersey/TestFullJAXWSRIAPIWithJAXWSClient.java:[20,29] package genealogy.client.cite does not exist
````

Same issue if I use the enunciate-full-tests profile with `mvn clean install -Penunciate-full-tests`

**Comment by stoicflame** [comment]
Fixed at 8d2a910.

Worked for me; let me know if you have issues with it.

**Comment by scoquelin** [comment]
Confirming that this is now working for me, thanks !
